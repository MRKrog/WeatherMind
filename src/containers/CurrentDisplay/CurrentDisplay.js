import React, { Component } from 'react';


export class CurrentDisplay extends Component {

  render() {
    return (
      <div className="CurrentDisplay">
        

        <div className="Today-Container">
          <p>{this.props.today.time}</p>
          <p>{this.props.today.tempHigh}</p>
          <p>{this.props.today.tempLow}</p>
        </div>

      </div>
    )
  }
}
