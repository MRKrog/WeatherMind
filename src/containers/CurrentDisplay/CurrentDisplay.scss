.CurrentDisplay {
  width: 100%;
  color:#fff;
  padding: 2em 1em .5em 1em;

  .CurrentWeather {
    display: flex;
    flex-direction: column;

    h3 {
      margin:0;
      font-size: 2.2em;
      font-weight: 300;
    }
    h4 {
      margin: 0;
      font-weight: 500;
      line-height: 1.5;
      font-size: .9em;
    }
    span {
      margin: 0;
      font-size: 6.5em;
      font-weight: 200;
      position: relative;
      line-height: 1.15;

      &::after {
        content: "\B0";
        font-size: .8em;
        top: 3px;
        line-height: 1;
        position: absolute;
      }
    }
  }

  .Today-Container {
    display: flex;
    justify-content: space-between;
    align-items: baseline;
    padding-top: 1.25em;

    .todayDay {
      margin: 0;
      span {
        text-transform: uppercase;
        padding-left: 1em;
      }
    }

    .tempLevels {
        display: flex;
      h6 {
        font-size: .9em;
      }
      h6:first-child {
        padding-right: 1em;
      }
      .tempHigh {
        margin:0;
      }
      .tempLow {
        margin:0;
      }
    }

  }

}
