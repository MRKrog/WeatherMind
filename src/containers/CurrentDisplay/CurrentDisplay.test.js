import React, { Component } from 'react';
import { CurrentDisplay, mapStateToProps } from './CurrentDisplay';
import { shallow } from 'enzyme';

const mockCurrent = {
  apparentTemperature: 73,
  humidity: 0.15,
  icon: "partly-cloudy-day",
  pressure: 1001.47,
  summary: "Mostly Cloudy",
  temperature: 73,
  time: 1555359672,
  uvIndex: 5,
  visibility: 6.34,
  windSpeed: 2.87,
}
const mockDetails = {
  city: "Denver",
  state: "Colorado",
  street: "1331 17th Street",
}
const mockToday = {
  humidity: 28.4,
  moonPhase: 37,
  precipProbability: 4 ,
  summary: "Mostly cloudy throughout the day.",
  sunriseTime: "6:23 AM",
  sunsetTime: "7:38 PM",
  tempHigh: 74,
  tempLow: 45,
  time: "Monday",
  uvIndex: 5,
  visibility: 6.75,
  windSpeed: 1.63,
}

describe('CurrentDisplay', () => {

  describe('CurrentDisplay Component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<CurrentDisplay current={mockCurrent} details={mockDetails} today={mockToday} />)
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    })

  })

  describe('mapStateToProps', () => {

    it('should return an object with the today weather object', () => {
      const mockState = {
        today: {
          humidity: 28.4,
          moonPhase: 37,
          precipProbability: 4 ,
        },
        filter: "SET_TODAY"
      }
      const expected = {
        today: {
          humidity: 28.4,
          moonPhase: 37,
          precipProbability: 4 ,
        },
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })

    it('should return an object with the current weather object', () => {
      const mockState = {
        current: {
          apparentTemperature: 73,
          humidity: 0.15,
          icon: "partly-cloudy-day",
        },
        filter: 'SET_CURRENT'
      }
      const expected = {
        current: {
          apparentTemperature: 73,
          humidity: 0.15,
          icon: "partly-cloudy-day",
        }
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })

    it('should return an object with the details weather object', () => {
      const mockState = {
        details: {
          city: "Denver",
          state: "Colorado",
          street: "1331 17th Street",
        },
        filter: "SET_DETAILS"
      }
      const expected = {
        details: {
          city: "Denver",
          state: "Colorado",
          street: "1331 17th Street",
        }
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
    
  })

})
