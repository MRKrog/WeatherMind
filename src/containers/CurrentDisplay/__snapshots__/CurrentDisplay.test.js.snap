// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentDisplay CurrentDisplay Component should match the snapshot 1`] = `
<div
  className="CurrentDisplay"
>
  <div
    className="CurrentWeather"
  >
    <h3>
      Denver
    </h3>
    <h4>
      Mostly Cloudy
    </h4>
    <span>
      73
    </span>
  </div>
  <div
    className="Today-Container"
  >
    <h5
      className="todayDay"
    >
      Monday
       
      <span>
        today
      </span>
    </h5>
    <section
      className="tempLevels"
    >
      <h6
        className="tempHigh"
      >
        74
      </h6>
      <h6
        className="tempLow"
      >
        45
      </h6>
    </section>
  </div>
</div>
`;
